gc/base/WorkPackets.hpp

```diff
@@ -110,6 +110,14 @@ class MM_WorkPackets: public MM_BaseVirtual
 
 	void reuseDeferredPackets(MM_EnvironmentBase *env);
 
+	/* methods used for Snapshot At the Beginning Barrier */
+	virtual MM_Packet *getBarrierPacket(MM_EnvironmentBase *env) { return NULL; }
+	virtual void removePacketFromInUseList(MM_EnvironmentBase *env, MM_Packet *packet) {}
+	virtual void putFullPacket(MM_EnvironmentBase *env, MM_Packet *packet) {}
+	virtual void putInUsePacket(MM_EnvironmentBase *env, MM_Packet *packet) {}
+	virtual void moveInUseToNonEmpty(MM_EnvironmentBase *env){}
+	virtual bool inUsePacketsAvailable(MM_EnvironmentBase *env){return false;}
+
```

Comment:
Do these methods need to be virtual? in RembemberSetWorkPackets, if you make _workPackets be a pointer in to more specific WorkPacketsSATB class, then I believe they don't. These methods should not be used in outside of SATB specific code.

Reply:
Agreed I think all of these could just be on the specific sublcass for SATB

Reply:
Moved these methods to WorkPacketsSATB and made _workPackets  pointer to be WorkPacketsSATB.